configure.ac

```diff
@@ -189,16 +192,70 @@ default=${localstatedir}/run/gstd/
 AC_ARG_WITH([gstd-runstatedir],
   [AS_HELP_STRING([--with-gstd-runstatedir],
     [Specify the location of the gstd's PID file @<:@default=${default}@:>@])],
-  [AC_SUBST(gstdrunstatedir, [$with_gstd_runstatedir])],
-  [AC_SUBST(gstdrunstatedir, [$default])])
+  [AC_SUBST(GSTD_RUN_STATE_DIR, [$with_gstd_runstatedir])],
+  [AC_SUBST(GSTD_RUN_STATE_DIR, [$default])])
 
 dnl allow the user to specify the location of the log files
 default=${localstatedir}/log/gstd/
 AC_ARG_WITH([gstd-logstatedir],
   [AS_HELP_STRING([--with-gstd-logstatedir],
     [Specify the location of gstd's log files @<:@default=${default}@:>@])],
-  [AC_SUBST([gstdlogstatedir], [$with_gstd_logstatedir])],
-  [AC_SUBST([gstdlogstatedir], [$default])])
+  [AC_SUBST([GSTD_LOG_STATE_DIR], [$with_gstd_logstatedir])],
+  [AC_SUBST([GSTD_LOG_STATE_DIR], [$default])])
+
+dnl allow the user to enable systemd gstd.service install
+AC_ARG_ENABLE(systemd,
+  AS_HELP_STRING([--enable-systemd], [Enable install of systemd gstd.service]),
+  [ systemd=true ],
+  [ systemd=false ])
+AM_CONDITIONAL(WITH_SYSTEMD, test "$systemd" = true)
+
+dnl if systemd option is enabled, check for libsystemd dependency
+if test "$systemd" = true; then
```

Comment:
I don't think this is necessary.

Reply:
But I do think we need to make sure the system supports systemd, maybe asking for a binary?

Reply:
I think the precedence should be:

1. If the user asks for systemd, enable it and fail if not supported.
2. If the user disables systemd, disable it.
3. If not specified, enable it by default if the system supports it.

Reply:
done